projectile.el

```diff
@@ -2654,6 +2654,65 @@ test/impl/other files as below:
   :type 'function
   :package-version '(projectile . "1.0.0"))
 
+(defun projectile--cmake-version()
+  "Compute CMake version."
+  (let* ((string (shell-command-to-string "cmake --version"))
+         (match (string-match "^cmake version \\([0-9]*\\)\.\\([0-9]*\\)" string)))
+    (when match
+      (mapcar 'string-to-number
+              (list (match-string 1 string)
+                    (match-string 2 string))))))
+
+(defun projectile--cmake-presets-supported()
+  "Check if CMake supports presets."
+  (and
+   (version-list-<= '(3 19) (projectile--cmake-version))))
+
+(defun projectile--cmake-read-presets(filename)
+  "Read CMake presets."
+  (when (file-exists-p filename)
+        (with-temp-buffer
+          (insert-file-contents filename)
+          (when (functionp 'json-parse-buffer)
+            (json-parse-buffer :array-type 'list)))))
+
+(defun projectile--cmake-read-configure-presets(filename)
+  "Get CMake presets from FILENAME."
+  (when-let ((presets (projectile--cmake-read-presets (projectile-expand-root filename))))
+    (gethash "configurePresets" presets)))
+
+(defun projectile--cmake-configure-presets()
+  "Get CMake presets"
+  (projectile-flatten
+   (mapcar 'projectile--cmake-read-configure-presets
+           '("CMakeUserPresets.json" "CMakeSystemPresets.json"))))
+
+(defconst projectile--cmake-no-preset "*no preset*")
+
+(defun projectile--cmake-select-configure-preset()
+  "Select CMake preset"
+  (if (and (projectile--cmake-presets-supported)
+           (functionp 'json-parse-buffer))
+    (let* ((configure-preset-names
+            (append `(,projectile--cmake-no-preset)
+                    (mapcar (lambda (configure-preset)
+                              (gethash "name" configure-preset))
+                            (projectile--cmake-configure-presets)))))
+      (projectile-completing-read "Use preset: "
+                                  configure-preset-names))
+    projectile--cmake-no-preset))
+
+(defun projectile--cmake-configure-cmd()
+  "Select CMake preset"
+  (interactive)
```

Comment:
If that's a private function it shouldn't not be an interactive command.

Reply:
Fixed.